File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int left = 0;
        int right = 0;
        int current = sequence[left];
        int currentMinLength = Integer.MAX_VALUE;
        int[] result = new int[2];
        
        while (left <= right && right < sequence.length) {
            
            if (current == k) {
                if (right - left < currentMinLength) {
                    currentMinLength = right - left;
                    result = new int[]{left, right};
                }
                current -= sequence[left];
                left++;
                right++;
                if (right < sequence.length) {
                    current += sequence[right];
                }
            } else if (current > k) {
                current -= sequence[left];
                left++;
            } else {
                right++;
                if (right < sequence.length) {
                    current += sequence[right];
                }
            }
        }
        
        return result;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    
    i = 0
    j = 0
    
    current = sequence[i]
    
    while i <= j and j < len(sequence):
        if current > k:
            current -= sequence[i]
            i += 1 
            
        elif current < k:
            j += 1
            if j < len(sequence):
                current += sequence[j]
            
        else:
            if not answer or answer[1] - answer[0] > j - i:
                answer = [i, j]
                
            current -= sequence[i]
            i += 1

        
    
    return answer